src/knowledge_sources/SegmentIdentityKS.m

```diff
@@ -12,8 +12,13 @@
             obj@AbstractAMLTTPKS( modelName, modelDir, ppRemoveDc );
             obj.setInvocationFrequency(4);
         end
+    end
+    
+    methods (Access = protected)
+        function prepAFEData( obj )
+        end
         
-        function execute( obj )
+        function amlttpExecute( obj )
             afeData = obj.getAFEdata();
             afeData = obj.blockCreator.cutDataBlock( afeData, obj.timeSinceTrigger );

```

Comment:
If amlttpExecute gets afeData as argument, these two lines can be saved here and also in NsrcsKs.